### PR TITLE
Fix AutoBatch with multispectral images

### PR DESCRIPTION
**Pull request by Y-T-G**
**MRE**
```
yolo train data=coco8-grayscale.yaml batch=-1
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds grayscale (non-3-channel) coverage to CUDA training tests and fixes AutoBatch to respect a model’s configured input channels 🧪⚡️

### 📊 Key Changes
- ✅ **Extended CUDA training tests** to run a quick training pass on `coco8-grayscale.yaml`, ensuring grayscale pipelines are exercised in CI.
- 🛠️ **Updated `autobatch()` to use the model’s configured input channels** (`model.yaml["channels"]`, defaulting to 3) instead of assuming RGB.
- 🔧 **AutoBatch profiling tensors now match real model inputs**, creating `torch.empty(b, ch, imgsz, imgsz)` rather than hardcoding `3` channels.

### 🎯 Purpose & Impact
- 📉 **Prevents AutoBatch mis-profiling and potential crashes** when training models/datasets with non-RGB inputs (e.g., grayscale medical/industrial imagery).
- 🎯 **Improves batch-size auto-selection accuracy** for custom channel configurations, leading to more reliable training setup on CUDA devices.
- 🧷 **Better regression protection**: CI will catch future changes that accidentally break grayscale or custom-channel training workflows.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review—an Ultralytics engineer will assist soon. 🧰✨

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

PR looks clean. The changes are safe and improve robustness for non-RGB inputs without introducing issues.

**Review by Laughing-q — APPROVED** [review]

**Comment by UltralyticsAssistant** [comment]
Merged — huge thanks @Y-T-G, with great contributions from @Laughing-q! This is a really impactful improvement for anyone training beyond standard RGB.

As Marie Curie put it: *“Nothing in life is to be feared, it is only to be understood.”* This PR adds that kind of understanding to our CI and AutoBatch pipeline—by exercising `coco8-grayscale.yaml` in CUDA tests and making `autobatch()` respect `model.yaml["channels"]`, you’ve turned a subtle edge case into a well-covered, reliable workflow.

Appreciate the care here: more accurate profiling tensors, fewer surprises for grayscale/custom-channel datasets, and stronger regression protection for the community. Thanks for helping make Ultralytics training more robust for real-world use cases.
